src/tests/nativeaot/SmokeTests/ComWrappers/ComWrappers.cs

```diff
@@ -0,0 +1,171 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if MULTIMODULE_BUILD && !DEBUG
+// Some tests won't work if we're using optimizing codegen, but scanner doesn't run.
+// This currently happens in optimized multi-obj builds.
+#define OPTIMIZED_MODE_WITHOUT_SCANNER
+#endif
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+
+// Name of namespace matches the name of the assembly on purpose to
+// ensure that we can handle this (mostly an issue for C++ code generation).
+namespace ComWrappersTests
+{
+    internal class Program
+    {
+        public static int Main(string[] args)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
I think if you put `<CLRTestTargetUnsupported Condition="'$(TargetsWindows)' != 'true'">true</CLRTestTargetUnsupported>` in the csproj, this test will be skipped. It's better than AOT compiling it only for the test to do nothing.